File: BookExamples/BroadcastReceiver/app/src/main/java/com/ebookfrenzy/broadcastreceiver/MyReceiver.java
package com.ebookfrenzy.broadcastreceiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

public class MyReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        // TODO: This method is called when the BroadcastReceiver is receiving
        // an Intent broadcast.

        String message = "Broadcast intent detected " + intent.getAction();
        Toast.makeText(context, message, Toast.LENGTH_LONG).show();
    }
}

File: class_book_examples/giraffe/chapter60_sendBroadcast/app/src/main/java/com/example/chapter60_sendbroadcast/MyReceiver.kt
package com.example.chapter60_sendbroadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.provider.SyncStateContract.Constants
import android.widget.Toast


class MyReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        // This method is called when the BroadcastReceiver is receiving an Intent broadcast.

        val myName: String? = intent.getStringExtra("myName")
        var message: String = "Broadcast " + intent.action +" no name sent"

        //NOT IN BOOK BUT THIS IS HOW YOU WOULD RECEIVE DATA USING A BROADCAST RECEIVER
        myName?.let{
            message = "Broadcast intent detected " + intent.action + " " + myName
        }

        Toast.makeText(context, message,Toast.LENGTH_LONG).show()
    }
}